File: src/main/java/lotto/utils/Validator.java
package lotto.utils;

import java.util.regex.Pattern;

import static lotto.domain.Lotto.LOTTO_NUM_COUNT;
import static lotto.utils.ErrorMessage.*;

public class Validator {

    private static final Pattern NUMBER_PATTERN = Pattern.compile("\\d+");

    public static void validateEmptyInput(String userInput) {
        if (userInput == null || userInput.isBlank()) {
            throw new IllegalArgumentException(EMPTY_INPUT_ERROR.getMessage());
        }
    }

    public static void validateNumberFormat(String userInput) {
        if (!NUMBER_PATTERN.matcher(userInput).matches()) {
            throw new IllegalArgumentException(NUMBER_FORMAT_ERROR.getMessage());
        }
    }

    public static void validateParsing(String[] userInput) {
        if (userInput.length != LOTTO_NUM_COUNT) {
            throw new IllegalArgumentException(LOTTO_NUM_COUNT_ERROR.getMessage());
        }
    }
}

File: src/main/kotlin/lotto/utils/Validator.kt
package lotto.utils

import lotto.domain.Lotto.Companion.LOTTO_NUM_COUNT
import java.util.regex.Pattern
import lotto.utils.ErrorMessage.*

object Validator {

    private val NUMBER_PATTERN: Pattern = Pattern.compile("\\d+")

    fun validateEmptyInput(userInput: String?) {
        if (userInput.isNullOrBlank()) {
            throw IllegalArgumentException(EMPTY_INPUT_ERROR.getMessage());
        }
    }

    fun validateNumberFormat(userInput: String?) {
        if(!NUMBER_PATTERN.matcher(userInput).matches()) {
            throw IllegalArgumentException(NUMBER_FORMAT_ERROR.getMessage());
        }
    }

    fun validateParsing(userInput: List<String>) {
        if (userInput.size != LOTTO_NUM_COUNT) {
            throw IllegalArgumentException(LOTTO_NUM_COUNT_ERROR.getMessage())
        }
    }

}
